packages/react-data-grid/src/ReactDataGrid.js

```diff
@@ -313,9 +317,6 @@ const ReactDataGrid = createReactClass({
   onCellCommit(commit: RowUpdateEvent) {
     let selected = Object.assign({}, this.state.selected);
     selected.active = false;
-    if (commit.key === 'Tab') {
-      selected.idx += 1;
```

Comment:
When `onCellCommit` it triggered by `Tab` (or `Enter` for that matter) the event is picked up by `onPressTab` (or `onPressEnter`) which already handles the navigation. They both are trying to move the `selected` state but since the `onPress` handler is called last it's state change "wins".

By only having the movement from a given key handled in one spot it makes it a little easier to follow. Also, this code wasn't respecting the boundaries of the grid so if you pressed `Tab` while editing the last cell in the row it would move the index to a non-existent column.